test/tests/rackhd20/test_rackhd20_api_workflows.py

```diff
@@ -69,5 +70,143 @@ def test_api_20_workflows_put_ipmi(self):
+                        "implementsTask": "Task.Base.Rest",
+                        "options": {
+                            "url": "http://localhost:8080/api/never/happen",
+                            "method": "GET"
+                        },
+                        "properties": {}
+                    }
+                },
+                {
+                    "label": "REST-call-final-good",
+                    "taskDefinition": {
+                        "friendlyName": "final REST call",
+                        "injectableName": "Task.Rest.GET",
+                        "implementsTask": "Task.Base.Rest",
+                        "options": {
+                            "url": "http://localhost:8080/api/1.1/nodes",
+                            "method": "GET"
+                        },
+                        "properties": {}
+                    },
+                    "waitOn": {
+                        "anyOf": {
+                            "REST-call-good": "succeeded",
+                            "REST-call-bad": "succeeded"
+                        },
+                        "REST-call-2": "succeeded"
+                    }
+                },
+                {
+                    "label": "REST-call-final-pending",
+                    "taskDefinition": {
+                        "friendlyName": "final REST call",
+                        "injectableName": "Task.Rest.GET",
+                        "implementsTask": "Task.Base.Rest",
+                        "options": {
+                            "url": "http://localhost:8080/api/1.1/nodes",
+                            "method": "GET"
+                        },
+                        "properties": {}
+                    },
+                    "waitOn": {
+                        "anyOf": {
+                            "REST-call-good": "failed",
+                            "REST-call-bad": "failed"
+                        }
+                    }
+                },
+                {
+                    "label": "REST-call-final-succeeded-2",
+                    "taskDefinition": {
+                        "friendlyName": "final REST call",
+                        "injectableName": "Task.Rest.GET",
+                        "implementsTask": "Task.Base.Rest",
+                        "options": {
+                            "url": "http://localhost:8080/api/1.1/nodes",
+                            "method": "GET"
+                        },
+                        "properties": {}
+                    },
+                    "waitOn": {
+                        "anyOf": {
+                            "REST-call-good": "failed",
+                            "REST-call-bad": "succeeded"
+                        }
+                    }
+                }
+            ]
+        }
+
+        node_id = fit_common.node_select()
+        if not node_id:
+            print "Cannot find any NODE connected with this RackHD server"
```

Comment:
Please remove the print statements and use logs method, see the fit template.
Since you stated with an existing script, add logging please.  :-)
import flogging
# set up the logging
logs = flogging.get_loggers()

logs.info("Cannot find....")
Or remove this section.  

Or do this check.
if not NODECATALOG: 
   self.fail("no nodes exist in nodelist")  - don't use raise please.